airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -435,7 +435,11 @@ def cast_property_type(self, property_name: str, column_name: str, jinja_column:
         elif is_number(definition["type"]):
             sql_type = jinja_call("dbt_utils.type_float()")
         elif is_timestamp_with_time_zone(definition):
-            sql_type = jinja_call("type_timestamp_with_timezone()")
+            sql_type = jinja_call(f"type_timestamp_with_timezone({jinja_column})")
+            if self.destination_type == DestinationType.SNOWFLAKE:
+                # snowflake uses case when statement to parse timestamp data
```

Comment:
Add more comments about `return` here, something like, `we return the modified sql_type statement instead of standard `f"cast({column_name} as {sql_type}) as {column_name}"`

Reply:
done